src/LayerTree/LayerTree.jsx

```diff
@@ -166,6 +167,10 @@ class LayerTree extends React.Component {
         this.rebuildTreeNodes();
       }
     }
+
+    if (nodeTitleRenderer != prevProps.nodeTitleRenderer){
```

Comment:
Equality check should be handled via [strict](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators) `!==` operator or lodash's `isEqual()` method.